File: C/Algorithms/Divide and Conquer/MinMax.c
#include<stdio.h>
int min,max;
int n;

void MinMax(int arr[], int s, int e){
	if(n == 1){
		return;
	}
	if(s == e){
		max = min = arr[s];
	}
	else if(s == e-1){
		if(arr[e] > arr[s]){
			min = arr[s];
			max = arr[e];
		}
		else{
			min = arr[e];
			max = arr[s];
		}
	}
	else{
		int m = s + (e-s)/2;
		MinMax(arr,s,m);
		int max1 = max,min1 = min;
		MinMax(arr,m+1,e);
		if(max1 > max){
			max = max1;
		}
		if(min1 < min){
			min = min1;
		}
	}
}


void main(){
	printf("Enter the number of elements\n");
	scanf("%d",&n);
	int arr[n];
	printf("Enter the array elements\n");
	for(int i=0;i<n;i++){
		scanf("%d",&arr[i]);
	}
	min = max = arr[0];
	MinMax(arr,0,n-1);
	printf("Min = %d\nMax = %d\n",min,max);
}

File: Python/Searching and Sorting/MinMax.py
min = 999999
max = 0

def minMax(lis,i,j):

    if j-i == 0:
        globals()['min'] = lis[i]
        globals()['max'] = lis[i]

    elif j-i == 1:
        if lis[i] > lis[j]:
            globals()['max'] = lis[i]
            globals()['min'] = lis[j]
        else:
            globals()['max'] = lis[j]
            globals()['min'] = lis[i]

    else:
        m = i + (j - i) // 2

        minMax(lis,i,m)
        l = globals()['max']
        s = globals()['min'] 

        minMax(lis,m,j)
        if s < min:
            globals()['min'] = s
        if l > max:
            globals()['max'] = l


lis = [2,1,4,5,17,3,9,10]
minMax(lis,0,len(lis)-1)
print("Min: {}, Max: {}".format(min,max))